addons/account/models/account_bank_statement_line.py

```diff
@@ -153,16 +135,45 @@ def default_get(self, fields_list):
 
     def init(self):
         super().init()
-        create_index(self.env.cr,
-                     indexname='account_bank_statement_line_internal_index_move_id_amount_idx',
-                     tablename='account_bank_statement_line',
-                     expressions=['internal_index', 'move_id', 'amount'],
-                     where='statement_id IS NULL')
+        create_index(  # used for default filters
+            self.env.cr,
+            indexname='account_bank_statement_line_unreconciled_idx',
+            tablename='account_bank_statement_line',
+            expressions=['journal_id', 'company_id', 'internal_index'],
+            where='NOT is_reconciled OR is_reconciled IS NULL',
+        )
+        create_index(  # used for the dashboard
+            self.env.cr,
+            indexname='account_bank_statement_line_orphan_idx',
+            tablename='account_bank_statement_line',
+            expressions=['journal_id', 'company_id', 'internal_index'],
+            where='statement_id IS NULL',
+        )
+        create_index(  # used in other cases
+            self.env.cr,
+            indexname='account_bank_statement_line_main_idx',
+            tablename='account_bank_statement_line',
+            expressions=['journal_id', 'company_id', 'internal_index'],
+        )
```

Comment:
Even if this index seems relatively heavy, it replaces 3 individual indexes on `journal_id`, `company_id`, and `internal_index`.
The keys are also fields that are frequently updated, usually just set upon initial INSERT and then not usually modified.
So the eventual overhead is just the updating of tuple-reference for non-HOT updates.

The other partial indexes are small or highly selective, low impacts on writes.